File: Data Structures & Algorithms/minimum-stack/submission-0.cpp
class MinStack {
public:
    std::stack<int> stack;
    std::stack<int> min;

    MinStack() {
        
    }
    
    void push(int val) {
        stack.push(val);
        if (min.empty() || val < min.top()) {
            min.push(val);
        } else {
            min.push(min.top());
        }
    }
    
    void pop() {
        stack.pop();
        min.pop();
    }
    
    int top() {
        return stack.top();
    }
    
    int getMin() {
        return min.top();
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] res = new int[temperatures.length];
        Stack<int[]> stack = new Stack<>();

        for (int i = 0; i < temperatures.length; i++) {
            int temp = temperatures[i];
            while (!stack.isEmpty() && temp > stack.peek()[0]) {
                int[] top = stack.pop();
                res[top[1]] = i - top[1];
            }
            stack.push(new int[]{temp, i});
        }

        return res;
    }
}
